Formula/libclang.rb

```diff
@@ -0,0 +1,139 @@
+class Libclang < Formula
+  desc "C interface to clang"
+  homepage "https://clang.llvm.org/doxygen/group__CINDEX.html"
+  url "https://releases.llvm.org/6.0.1/llvm-6.0.1.src.tar.xz"
+  sha256 "b6d6c324f9c71494c0ccaf3dac1f16236d970002b42bb24a6c9e1634f7d0f4e2"
+
+  keg_only :provided_by_macos
+
+  depends_on "cmake" => :build
+
+  resource "clang" do
+    url "https://releases.llvm.org/6.0.1/cfe-6.0.1.src.tar.xz"
+    sha256 "7c243f1485bddfdfedada3cd402ff4792ea82362ff91fbdac2dae67c6026b667"
+  end
+
+  # According to the official llvm readme, GCC 4.7+ is required
+  fails_with :gcc_4_0
+  fails_with :gcc
+  ("4.3".."4.6").each do |n|
+    fails_with :gcc => n
+  end
```

Comment:
All those versions of GCC are very old and not carried by recent macOS and recent Homebrew anymore. Remove the whole block.